Content.Shared/ReagentOnItem/Systems/SpaceLubeOnItemSystem.cs

```diff
@@ -0,0 +1,114 @@
+using Content.Shared.IdentityManagement;
+using Content.Shared.Popups;
+using Content.Shared.Throwing;
+using Robust.Shared.Random;
+using Content.Shared.Clothing.Components;
+using Content.Shared.Inventory;
+using Content.Shared.Examine;
+using Robust.Shared.GameStates;
+using Content.Shared.Hands;
+using Content.Shared.NameModifier.EntitySystems;
+
+namespace Content.Shared.ReagentOnItem;
+
+public sealed class SpaceLubeOnItemSystem : EntitySystem
+{
+    [Dependency] private readonly ThrowingSystem _throwing = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly InventorySystem _inventory = default!;
+    [Dependency] private readonly SharedTransformSystem _xform = default!;
+    [Dependency] private readonly NameModifierSystem _nameMod = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<SpaceLubeOnItemComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<SpaceLubeOnItemComponent, GotAttemptedHandPickupEvent>(OnHandPickUp);
+        SubscribeLocalEvent<SpaceLubeOnItemComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<SpaceLubeOnItemComponent, RefreshNameModifiersEvent>(OnRefreshNameModifiers);
+
+        SubscribeLocalEvent<SpaceLubeOnItemComponent, ComponentGetState>(GetSpaceLubeState);
+        SubscribeLocalEvent<SpaceLubeOnItemComponent, ComponentHandleState>(HandleSpaceLubeState);
+    }
+
+    private void OnInit(EntityUid uid, SpaceLubeOnItemComponent component, ComponentInit args)
+    {
+        _nameMod.RefreshNameModifiers(uid);
+    }
+
+    private void OnHandPickUp(EntityUid uid, SpaceLubeOnItemComponent component, GotAttemptedHandPickupEvent args)
+    {
+        if (args.Cancelled)
+            return;
+
+        var handContainer = args.Hand.Container;
+        if (handContainer == null)
+            return;
+
+        var entityWhoPickedUp = handContainer.Owner;
+
+        _inventory.TryGetSlotEntity(entityWhoPickedUp, "gloves", out var gloves);
+
+        if (HasComp<NonStickSurfaceComponent>(gloves))
+            return;
+
+        if (component.EffectStacks < 1)
+        {
+            RemComp<SpaceLubeOnItemComponent>(uid);
+            _nameMod.RefreshNameModifiers(uid);
+            return;
+        }
+
+        args.Cancel();
+
+        var randDouble = _random.NextDouble();
+        if (randDouble > 1 - component.ChanceToDecreaseReagentOnGrab)
+        {
+            component.EffectStacks -= 1;
+        }
+
+        Dirty(uid, component);
+
+        var entTransform = Transform(entityWhoPickedUp);
+        var xformQuery = GetEntityQuery<TransformComponent>();
+        var worldPos = _xform.GetWorldPosition(entTransform, xformQuery);
+        var vecFromObjectToEnt = worldPos - _xform.GetWorldPosition(uid, xformQuery);
```

Comment:
These methods are marked as obsolete, you're also getting xformQuery by hand here but using Transform anyway.